torch/_tensor.py

```diff
@@ -378,9 +378,18 @@ def _reduce_ex_internal(self, proto):
             )
             return (torch._utils._rebuild_nested_tensor, args_nested)
         elif (
-            self.data_ptr() == 0
-            and type(self) is not torch.Tensor
```

Comment:
please just re-order this check and the one above isntead of special casing!

Reply:
Was reordered